ui/src/timeMachine/components/variableToolbar/FetchVariables.tsx

```diff
@@ -0,0 +1,50 @@
+// Libraries
+import {PureComponent} from 'react'
+import _ from 'lodash'
+
+// APIs
+import {client} from 'src/utils/api'
+
+// Types
+import {RemoteDataState} from 'src/types'
+import {Variable} from '@influxdata/influx'
+
+interface Props {
+  children: (variables: Variable[], loading: RemoteDataState) => JSX.Element
+}
+
+interface State {
+  variables: Variable[]
+  loading: RemoteDataState
+}
+
+class FetchVariables extends PureComponent<Props, State> {
+  constructor(props: Props) {
```

Comment:
total nit, but there's a slightly more declarative way to specify initial state:

```
class FetchVariables extends PureComponent<Props, State> {
  public state: State = {
    variables: [],
    loading: RemoteDataState.NotStarted,
  }
```